test/python/test_jacobian.py

```diff
@@ -702,3 +870,275 @@ def setUpClass(cls):
+            - name: gas
+              thermo: ideal-gas
+              species:
+              - gri30.yaml/species: [H2, H2O, H2O2, O2]
+              kinetics: gas
+              reactions:
+              - gri30.yaml/reactions: declared-species
+              skip-undeclared-third-bodies: true
+            - name: Pt_surf
+              thermo: ideal-surface
+              species:
+              - ptcombust.yaml/species: [PT(S), H(S), H2O(S), OH(S), O(S)]
+              kinetics: surface
+              reactions: [ptcombust.yaml/reactions: declared-species]
+              site-density: 3e-09
+        """
+        # create phase objects
+        cls.gas = ct.Solution(yaml=phase_defs, name="gas")
+        cls.surf = ct.Interface(yaml=phase_defs, name="Pt_surf", adjacent=[cls.gas])
+        cls.gas.TPX = 800, 2*ct.one_atm, "H2:1.5, O2:1.0, H2O2:0.75, H2O:0.3"
+        cls.surf.TPX = 800, 2*ct.one_atm , "PT(S):4.0, H(S):0.5, H2O(S):0.1, OH(S):0.2, O(S):0.8"
+        cls.gas_tpx = cls.gas.TPX
+        cls.surf_tpx = cls.surf.TPX
+        super().setUpClass()
+
+class SurfInterfaceArrhenius(PlatinumHydrogen, utilities.CanteraTest):
+    rxn_idx = 7
+    equation = "H(S) + O(S) <=> OH(S) + PT(S)"
+    rate_type = "interface-Arrhenius"
+
+class SurfGasFwdStickingArrhenius(PlatinumHydrogen, utilities.CanteraTest):
+    rxn_idx = 5
+    equation = "H2O + PT(S) => H2O(S)"
+    rate_type = "sticking-Arrhenius"
+
+class SurfGasInterfaceArrhenius(PlatinumHydrogen, utilities.CanteraTest):
+    rxn_idx = 0
+    equation = "H2 + 2 PT(S) => 2 H(S)"
+    rate_type = "interface-Arrhenius"
+    orders = {"PT(S)": 1, "H2": 1, "H(S)": 2}
+
+class GasSurfInterfaceArrhenius(PlatinumHydrogen, utilities.CanteraTest):
+    rxn_idx = 6
+    equation = "H2O(S) => H2O + PT(S)"
+    rate_type = "interface-Arrhenius"
+
+class SurfaceFullTests:
+    # Generic test class to check derivatives evaluated for an entire reaction mechanisms
+    rtol = 1e-4
+
+    @classmethod
+    def setUpClass(cls):
+        # all species indexs
+        all_species = cls.surf.species() + cls.gas.species()
+        cls.sidxs  = {spec.name:i for i, spec in enumerate(all_species)}
+
+    def setUp(self):
+        # gas phase
+        self.gas.TPX = self.gas_tpx
+        self.gas.derivative_settings = {} # reset defaults
+        # surface phase
+        self.surf.TPX = self.surf_tpx
+        self.surf.derivative_settings = {"skip-coverage-dependence": True, "skip-electrochemistry": True}
+
+    # closure to get concentrations vector
+    def get_concentrations(self):
+        return np.concatenate((self.surf.concentrations, self.gas.concentrations))
+
+    def test_forward_rop_ddCi(self):
+        # check forward rop against analytical result
+        drop = self.surf.forward_rates_of_progress_ddCi
+        rop = self.surf.forward_rates_of_progress
+        conc = self.get_concentrations()
+        # multiply derivatives with concentrations
+        drop = np.matmul(drop, conc)
```

Comment:
Here and in a couple of other locations, you can write
```suggestion
        drop = drop @ conc
```